shebang.go

```diff
@@ -0,0 +1,92 @@
+package slinguist
+
+import (
+	"bufio"
+	"bytes"
+	"regexp"
+	"strings"
+)
+
+var (
+	shebangExecHack = regexp.MustCompile(`exec (\w+).+\$0.+\$@`)
+	pythonVersion   = regexp.MustCompile(`python\d\.\d+`)
+)
+
+func GetLanguageByShebang(data []byte) (lang string, safe bool) {
+	interpreter := getInterpreter(data)
+	lang = OtherLanguage
+	if langs, ok := languagesByInterpreter[interpreter]; ok {
+		lang = langs[0]
+		safe = len(langs) == 1
+	}
+
+	return
```

Comment:
Like this function is used as a strateg, you can pass a language to the function that is not a interpreted language so is not in languagesByInterpreter. Then safe must be false to fall to the next strategy.